File: app/src/main/java/com/huajiqaq/sgamehook/XposedMod.java
package com.huajiqaq.sgamehook;

import android.util.Log;
import de.robv.android.xposed.IXposedHookLoadPackage;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

public class XposedMod implements IXposedHookLoadPackage {
    private static final String TAG = "XPOSED_SGAMEHOOK";

    @Override
    public void handleLoadPackage(XC_LoadPackage.LoadPackageParam lpparam) {
        String packageName = lpparam.packageName;
        String processName = lpparam.processName;

        if ((packageName.equals("com.tencent.tmgp.sgame") || packageName.equals("com.tencent.tmgp.sgamece")) && packageName.equals(processName)) {
            Log.d(TAG, "Package detected: " + packageName + ", Process: " + processName);
            System.loadLibrary("sgame_hook");
        }
    }
}

File: app/src/main/java/com/huajiqaq/sgamehook/XposedMod.kt
package com.huajiqaq.sgamehook

import de.robv.android.xposed.IXposedHookLoadPackage
import de.robv.android.xposed.callbacks.XC_LoadPackage
import android.util.Log

class XposedMod : IXposedHookLoadPackage {
    private val TAG = "SGAMEHOOK"

    override fun handleLoadPackage(lpparam: XC_LoadPackage.LoadPackageParam) {
        val packageName = lpparam.packageName
        val processName = lpparam.processName

        if ((packageName == "com.tencent.tmgp.sgame" || packageName == "com.tencent.tmgp.sgamece") && packageName == processName) {
            Log.d(TAG, "Package detected: $packageName, Process: $processName")
            System.loadLibrary("sgame_hook");
        }
    }

}